valhalla/sif/dynamiccost.h

```diff
@@ -204,7 +204,9 @@ class DynamicCost {
    * @return  Returns true if access is allowed, false if not.
    */
   inline virtual bool Allowed(const baldr::NodeInfo* node) const {
-    return (node->access() & access_mask_) || ignore_access_;
+    return ((node->access() & access_mask_) &&
+            !(exclude_cash_only_tolls_ && node->cash_only_toll())) ||
+           ignore_access_;
```

Comment:
this doesnt look right, i think it should be:

```c++
return ((node->access() & access_mask_) || ignore_access_) &&
            !(exclude_cash_only_tolls_ && node->cash_only_toll());
```

basically the ignoring access has to do with the access and the cash toll is a separate concern

cc @gknisely @ktatterso

Reply:
@kevinkreiser ty